ip/addr.py

```diff
@@ -127,6 +117,28 @@ def ifupdown(iface, up):
                          % iface)
 
 
+def dhclient(iface, enable, script=None):
+    # Disable the dhcp client and flush interface
+    try:
+        dhclients = sh.awk(
+            sh.grep(
+                sh.grep(sh.ps("ax"), iface, _timeout=5),
+                "dhclient", _timeout=5),
+            "{print $1}")
+        dhclients = dhclients.split()
+        for dhclient in dhclients:
+            sh.kill(dhclient)
+    except Exception as e:
+        _logger.info("Failed to stop dhclient: %s" % e)
+        pass
+
+    if enable:
+        if script:
+            sh.dhclient("-nw", "-sf", script, iface, _bg=True)
+        else:
+            sh.dhclient("-nw", iface, _bg=True)
```

Comment:
yeah, unless dhclient will do technique like `fork, fork` change its parent pid to 1.